app/libs/destroy_clusters.py

```diff
@@ -0,0 +1,267 @@
+import multiprocessing
+import os
+import re
+import shutil
+from http import HTTPStatus
+from pathlib import Path
+
+import click
+import yaml
+from clouds.aws.session_clients import s3_client
+from libs.aws_ipi_clusters import (
+    create_or_destroy_aws_ipi_cluster,
+    download_openshift_install_binary,
+)
+from libs.rosa_clusters import rosa_delete_cluster
+from utils.const import AWS_STR, CLUSTER_DATA_YAML_FILENAME
+
+S3_EXTRACTED_DATA_FILES_DIR_NAME = "extracted_clusters_files"
+
+
+def download_and_extract_s3_file(
+    client, bucket, bucket_filepath, target_dir, target_filename, extracted_target_dir
+):
+    target_file_path = os.path.join(target_dir, target_filename)
+    click.echo(f"Download {bucket_filepath} from {bucket} bucket to {target_file_path}")
+    try:
+        client.download_file(
+            Bucket=bucket, Key=bucket_filepath, Filename=target_file_path
+        )
+
+        target_extract_dir = os.path.join(extracted_target_dir, target_filename)
+        click.echo(
+            f"Extract {target_filename} from {target_file_path} to {target_extract_dir}"
+        )
+        shutil.unpack_archive(
+            filename=target_file_path,
+            extract_dir=target_extract_dir,
+            format="zip",
+        )
+    except client.exceptions.ClientError as ex:
+        if ex.response["Error"]["Code"] == str(HTTPStatus.NOT_FOUND):
+            click.echo(f"{bucket_filepath} not found in {bucket}")
+            raise click.Abort()
+
+
+def _destroy_clusters_download_installer_binary(
```

Comment:
The name is confusing, when reading this I this you will destroy the binaries directories.